### PR TITLE
feat(remix): add remix

### PR DESCRIPTION
**Pull request by Coly010**
## Current Behavior

Remix support currently lives in `nx-labs`

## Expected Behavior

Remix support should be in `nx`

## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by Moxita33 — APPROVED** [review]

**Review by isaacplmann — APPROVED** [review]

**Review by FrozenPandaz — DISMISSED** [review]

File: packages/remix/src/generators/loader/loader.impl.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+  if (!tree.exists(routeFilePath)) {
+    throw new Error(
+      `Route path does not exist: ${routeFilePath}. Please generate a Remix route first.`
+    );
+  }
+
+  insertImport(tree, routeFilePath, 'useLoaderData', '@remix-run/react');
+  insertImport(tree, routeFilePath, 'json', '@remix-run/node');
+  insertImport(tree, routeFilePath, 'LoaderArgs', '@remix-run/node', {
```
Comment (nartc) [inline packages/remix/src/generators/loader/loader.impl.ts:22]: issue/nit: `LoaderArgs` is renamed to `LoaderFunctionArgs` in v2

**Review by jaysoo — APPROVED** [review]

**Review by mandarini — APPROVED** [review]

File: e2e/remix/tests/nx-remix.test.ts [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+  uniq,
+  updateFile,
+  runCommandAsync,
+} from '@nx/e2e/utils';
+
+describe('remix e2e', () => {
+  let proj: string;
+
+  beforeAll(() => {
+    proj = newProject();
```
Comment (meeroslav) [inline e2e/remix/tests/nx-remix.test.ts:17]: You probably would like to follow the convention that @FrozenPandaz recommended on adding just the packages we need for those tests.

File: packages/remix/package.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  "executors": "./executors.json",
+  "nx-migrations": {
+    "migrations": "./migrations.json"
+  },
+  "dependencies": {
+    "@nx/devkit": "file:../devkit",
+    "@nx/js": "file:../js",
+    "@nx/react": "file:../react",
+    "@nx/vite": "file:../vite",
+    "@nx/jest": "file:../jest",
```
Comment (meeroslav) [inline packages/remix/package.json:35]: Why are jest, cypress, vite and storybook dependencies of remix? Those should be optional steps using `ensurePackage` when needed.

File: packages/remix/.eslintrc.json [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        "@nx/dependency-checks": [
+          "error",
+          {
+            "buildTargets": ["build-base"],
+            "ignoredDependencies": [
+              "nx",
+              "typescript",
+              "@nx/cypress",
+              "@nx/playwright",
+              "@nx/storybook"
```
Comment (meeroslav) [inline packages/remix/.eslintrc.json:39]: These are not up to date. For example, `@nx/cypress` is added to `package.json`. I don't see `eslint` here or in `package.json`. How is linting being added?

**Review by meeroslav — CHANGES_REQUESTED** [review]
Linting seems to be missing for remix application.

Also package.json and .eslintrc don't seem to be optimally configured.

File: packages/remix/project.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+  "sourceRoot": "packages/remix/src",
+  "projectType": "library",
+  "targets": {
+    "build": {
+      "command": "node ./scripts/copy-readme.js remix",
+      "outputs": ["{workspaceRoot}/build/packages/remix"]
+    },
+    "lint": {
+      "executor": "@nx/eslint:lint",
+      "outputs": ["{options.outputFile}"],
```
Comment (meeroslav) [inline packages/remix/project.json:13]: Remove outputs here. This is no longer generated automatically

Comment (meeroslav) [reply packages/remix/project.json:13]: Actually the entire lint config should be gone, use just `"lint": {}`

File: packages/remix/project.json [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+      "options": {
+        "lintFilePatterns": ["packages/remix/**/*.ts"]
+      }
+    },
+    "test": {
+      "executor": "@nx/jest:jest",
```
Comment (meeroslav) [inline packages/remix/project.json:19]: I believe this should also be used from targetDefaults

**Review by meeroslav — APPROVED** [review]
Approved with small comment

**Review by meeroslav — APPROVED** [review]
Approved with small comment

**Review by AgentEnder — APPROVED** [review]

**Pull request merged**